discovery/gossiper_test.go

```diff
@@ -667,12 +690,23 @@ func (ctx *testCtx) createAnnouncementWithoutProof(blockHeight uint32,
 		Hash:  fundingTx.TxHash(),
 		Index: 0,
 	}
-	ctx.chain.addUtxo(chanUtxo, tx)
 
-	fundingBlock := &wire.MsgBlock{
-		Transactions: []*wire.MsgTx{fundingTx},
+	if opts.modifier != edgeCreationSpentUTXO {
```

Comment:
feels like we are adding some logic branches to the unit tests, which IMO is usually bad as they are difficult to maintain. Tho I think it's pre-exisiting, so just wanna mention this for future unit tests endeavors.

Reply:
totally agree. 

let me know what you think of the latest iteration which uses your mock idea. 

it still does a biiiit of logic branching to prep the mock chain calls in a specific way but i think we cant really get around that. See the next PR for how the prep here is used: https://github.com/lightningnetwork/lnd/pull/9478